## Analyze A/B Test Results

You may either submit your notebook through the workspace here, or you may work from your local machine and submit through the next page.  Either way assure that your code passes the project [RUBRIC](https://review.udacity.com/#!/projects/37e27304-ad47-4eb0-a1ab-8c12f60e43d0/rubric).  **Please save regularly.**

This project will assure you have mastered the subjects covered in the statistics lessons.  The hope is to have this project be as comprehensive of these topics as possible.  Good luck!

## Table of Contents
- [Introduction](#intro)
- [Part I - Probability](#probability)
- [Part II - A/B Test](#ab_test)
- [Part III - Regression](#regression)


<a id='intro'></a>
### Introduction

A/B tests are very commonly performed by data analysts and data scientists.  It is important that you get some practice working with the difficulties of these 

For this project, you will be working to understand the results of an A/B test run by an e-commerce website.  Your goal is to work through this notebook to help the company understand if they should implement the new page, keep the old page, or perhaps run the experiment longer to make their decision.

**As you work through this notebook, follow along in the classroom and answer the corresponding quiz questions associated with each question.** The labels for each classroom concept are provided for each question.  This will assure you are on the right track as you work through the project, and you can feel more confident in your final submission meeting the criteria.  As a final check, assure you meet all the criteria on the [RUBRIC](https://review.udacity.com/#!/projects/37e27304-ad47-4eb0-a1ab-8c12f60e43d0/rubric).

<a id='probability'></a>
#### Part I - Probability

To get started, let's import our libraries.

In [ ]:
import pandas as pd
import numpy as np
import random
import matplotlib.pyplot as plt
%matplotlib inline
#We are setting the seed to assure you get the same answers on quizzes as we set up
random.seed(42)

`1.` Now, read in the `ab_data.csv` data. Store it in `df`.  **Use your dataframe to answer the questions in Quiz 1 of the classroom.**

a. Read in the dataset and take a look at the top few rows here:

In [ ]:
df = pd.read_csv('ab_data.csv')
df.head()

b. Use the cell below to find the number of rows in the dataset.

In [ ]:
df.shape

c. The number of unique users in the dataset.

In [ ]:
df.nunique()

d. The proportion of users converted.

In [ ]:
df.query('converted == 1').shape[0]/df.shape[0]

e. The number of times the `new_page` and `treatment` don't match.

In [ ]:
df[((df['group'] == "treatment") & 
    (df['landing_page'] != "new_page")) |
   ((df['group'] == "control") & (df['landing_page'] != "old_page"))].shape[0]
    

f. Do any of the rows have missing values?

In [ ]:
df.isnull().values.any()

`2.` For the rows where **treatment** does not match with **new_page** or **control** does not match with **old_page**, we cannot be sure if this row truly received the new or old page.  Use **Quiz 2** in the classroom to figure out how we should handle these rows.  

a. Now use the answer to the quiz to create a new dataset that meets the specifications from the quiz.  Store your new dataframe in **df2**.

In [ ]:
df2 = df[((df['group'] == "treatment") & 
    (df['landing_page'] == "new_page")) |
   ((df['group'] == "control") & (df['landing_page'] == "old_page"))]
df2.shape[0]

In [ ]:
# Double Check all of the correct rows were removed - this should be 0
df2[((df2['group'] == 'treatment') == (df2['landing_page'] == 'new_page')) == False].shape[0]

`3.` Use **df2** and the cells below to answer questions for **Quiz3** in the classroom.

a. How many unique **user_id**s are in **df2**?

In [ ]:
df2.nunique()

b. There is one **user_id** repeated in **df2**.  What is it?

In [ ]:
df2['duplicated'] = df2['user_id'].duplicated()
df2[df2['duplicated'] == True]

c. What is the row information for the repeat **user_id**? 

In [ ]:
df2[df2['duplicated'] == True]

d. Remove **one** of the rows with a duplicate **user_id**, but keep your dataframe as **df2**.

In [ ]:
df2 = df2.drop_duplicates(subset = "user_id")
df2 = df2.drop(['duplicated'], axis = 1)

In [ ]:
df2.nunique()
df2.head()

`4.` Use **df2** in the cells below to answer the quiz questions related to **Quiz 4** in the classroom.

a. What is the probability of an individual converting regardless of the page they receive?

In [17]:
df2[df2['converted'] == 1].shape[0]/df2.shape[0]

0.11959708724499628

b. Given that an individual was in the `control` group, what is the probability they converted?

In [18]:
df2[(df2['group'] == "control") & (df2['converted'] == 1)].shape[0]/df2[df2['group'] == "control"].shape[0]

0.1203863045004612

c. Given that an individual was in the `treatment` group, what is the probability they converted?

In [19]:
df2[(df2['group'] == "treatment") & (df2['converted'] == 1)].shape[0]/df2.shape[0]

0.05941139223081794

d. What is the probability that an individual received the new page?

In [20]:
df2[df2['landing_page'] == "new_page"].shape[0]/df2.shape[0]

0.5000619442226688

e. Consider your results from parts (a) through (d) above, and explain below whether you think there is sufficient evidence to conclude that the new treatment page leads to more conversions.

Based on the result above I strongly believe that new treatment page lead to more conversions, since the number of people who converted from "control" group is twice as high as "Treatment" group

<a id='ab_test'></a>
### Part II - A/B Test

Notice that because of the time stamp associated with each event, you could technically run a hypothesis test continuously as each observation was observed.  

However, then the hard question is do you stop as soon as one page is considered significantly better than another or does it need to happen consistently for a certain amount of time?  How long do you run to render a decision that neither page is better than another?  

These questions are the difficult parts associated with A/B tests in general.  


`1.` For now, consider you need to make the decision just based on all the data provided.  If you want to assume that the old page is better unless the new page proves to be definitely better at a Type I error rate of 5%, what should your null and alternative hypotheses be?  You can state your hypothesis in terms of words or in terms of **$p_{old}$** and **$p_{new}$**, which are the converted rates for the old and new pages.

Null: **$p_{new}$** - **$p_{old}$** <= 0  
Alternative: **$p_{new}$** - **$p_{old}$** > 0

`2.` Assume under the null hypothesis, $p_{new}$ and $p_{old}$ both have "true" success rates equal to the **converted** success rate regardless of page - that is $p_{new}$ and $p_{old}$ are equal. Furthermore, assume they are equal to the **converted** rate in **ab_data.csv** regardless of the page. <br><br>

Use a sample size for each page equal to the ones in **ab_data.csv**.  <br><br>

Perform the sampling distribution for the difference in **converted** between the two pages over 10,000 iterations of calculating an estimate from the null.  <br><br>

Use the cells below to provide the necessary parts of this simulation.  If this doesn't make complete sense right now, don't worry - you are going to work through the problems below to complete this problem.  You can use **Quiz 5** in the classroom to make sure you are on the right track.<br><br>

a. What is the **conversion rate** for $p_{new}$ under the null? 

In [21]:
p_new = round(df2.query('converted == 1').shape[0]/df2.shape[0],4)
p_new

0.1196

b. What is the **conversion rate** for $p_{old}$ under the null? <br><br>

In [22]:
p_old = round(df2.query('converted == 1').shape[0]/df2.shape[0],4)
p_old

0.1196

c. What is $n_{new}$, the number of individuals in the treatment group?

In [23]:
n_new = df2.query('landing_page == "new_page" and group == "treatment"').shape[0]
n_new

145310

d. What is $n_{old}$, the number of individuals in the control group?

In [24]:
n_old = df2.query('landing_page == "old_page"').shape[0]
n_old

145274

e. Simulate $n_{new}$ transactions with a conversion rate of $p_{new}$ under the null.  Store these $n_{new}$ 1's and 0's in **new_page_converted**.

In [25]:
new_page_converted = np.random.binomial(n_new,p_new)
new_page_converted

17272

f. Simulate $n_{old}$ transactions with a conversion rate of $p_{old}$ under the null.  Store these $n_{old}$ 1's and 0's in **old_page_converted**.

In [26]:
old_page_converted = np.random.binomial(n_old,p_old)
old_page_converted

17157

g. Find $p_{new}$ - $p_{old}$ for your simulated values from part (e) and (f).

In [27]:
obs_diff = new_page_converted/n_new - old_page_converted/n_old
obs_diff

0.00076215239940991

h. Create 10,000 $p_{new}$ - $p_{old}$ values using the same simulation process you used in parts (a) through (g) above. Store all 10,000 values in a NumPy array called **p_diffs**.

In [ ]:
p_diffs = []

#using binominal to get the answer
#I have changed my code based on reveiwer's feedback, however even though this definetely made my code faster for this box, when it comes to graphing my histogram it is taking me reasonably long time to execute my code.
new_page_converted = np.random.binomial(n_new,p_new,10000)
old_page_converted = np.random.binomial(n_old,p_old,10000)
act_diff = new_page_converted/n_new - old_page_converted/n_old
p_diffs.append(act_diff)

i. Plot a histogram of the **p_diffs**.  Does this plot look like what you expected?  Use the matching problem in the classroom to assure you fully understand what was computed here.

In [ ]:
p_diffs = np.array(p_diffs)
plt.hist(p_diffs)
plt.axvline(x=obs_diff, color="red")


j. What proportion of the **p_diffs** are greater than the actual difference observed in **ab_data.csv**?

In [29]:
act_diffs = df2.query('group == "treatment"')['converted'].mean() - df2.query('group == "control"')['converted'].mean()
(p_diffs > act_diffs).mean()

0.90310000000000001

k. Please explain using the vocabulary you've learned in this course what you just computed in part **j.**  What is this value called in scientific studies?  What does this value mean in terms of whether or not there is a difference between the new and old pages?

On part 'j' we have computed the p - value of the given example. and since the given p - value is higher than 0.05, we can say that we will fail to reject the null hypothesis where p(new) <= p(old)

l. We could also use a built-in to achieve similar results.  Though using the built-in might be easier to code, the above portions are a walkthrough of the ideas that are critical to correctly thinking about statistical significance. Fill in the below to calculate the number of conversions for each page, as well as the number of individuals who received each page. Let `n_old` and `n_new` refer the the number of rows associated with the old page and new pages, respectively.

In [40]:
import statsmodels.api as sm
from scipy import stats
stats.chisqprob = lambda chisq, df: stats.chi2.sf(chisq, df)

convert_old = df2.query('landing_page == "old_page"')['converted'].sum()
convert_new = df2.query('landing_page == "new_page"')['converted'].sum()
n_old = df2.query('landing_page == "old_page"').shape[0]
n_new = df2.query('landing_page == "new_page"').shape[0]

m. Now use `stats.proportions_ztest` to compute your test statistic and p-value.  [Here](https://docs.w3cub.com/statsmodels/generated/statsmodels.stats.proportion.proportions_ztest/) is a helpful link on using the built in.

In [49]:
#based on the reviewer's feedback I have added 'alternative = smaller' into my function
#this is because the propbability of people who are converting from old page is higher than people who convert from new page.
z_score, p_value = sm.stats.proportions_ztest([convert_old, convert_new],[n_old, n_new], alternative="smaller" )
z_score, p_value

(1.3109241984234394, 0.90505831275902449)

In [34]:
#culmulative distribution function
stats.norm.cdf(z_score)
#probability of density of function in 95% ratecdfcd
stats.norm.ppf(1-(0.05/2))

1.959963984540054

n. What do the z-score and p-value you computed in the previous question mean for the conversion rates of the old and new pages?  Do they agree with the findings in parts **j.** and **k.**?

As stated in 'j' and 'k' p - value represents whether we can fail to reject the null hypothesis, and since this value closely matches the value above, we can say that we can "fail to reject the null hypothesis". 
Z - score measures number of std a given value x falls from the mean, since z_score falls within the denstiy of function, we can we that z_score and p_value matches.

<a id='regression'></a>
### Part III - A regression approach

`1.` In this final part, you will see that the result you achieved in the A/B test in Part II above can also be achieved by performing regression.<br><br> 

a. Since each row is either a conversion or no conversion, what type of regression should you be performing in this case?

Since I will be working with binary (0/1) answer, I can either use Simple Linear Model or Logstic Regressino to predict whether the user will convert using the 'new_page.'

b. The goal is to use **statsmodels** to fit the regression model you specified in part **a.** to see if there is a significant difference in conversion based on which page a customer receives. However, you first need to create in df2 a column for the intercept, and create a dummy variable column for which page each user received.  Add an **intercept** column, as well as an **ab_page** column, which is 1 when an individual receives the **treatment** and 0 if **control**.

In [36]:
df2['intercept'] = 1
df2[['new_page','old_page']] = pd.get_dummies(df2['landing_page'])
df2[['control','ab_page']] = pd.get_dummies(df2['group'])
df2.drop(['new_page', 'control' ], axis=1, inplace=True)
df2.head()

user_id                   timestamp      group landing_page  converted  \
0   851104  2017-01-21 22:11:48.556739    control     old_page          0   
1   804228  2017-01-12 08:01:45.159739    control     old_page          0   
2   661590  2017-01-11 16:55:06.154213  treatment     new_page          0   
3   853541  2017-01-08 18:28:03.143765  treatment     new_page          0   
4   864975  2017-01-21 01:52:26.210827    control     old_page          1   

   intercept  old_page  ab_page  
0          1         1        0  
1          1         1        0  
2          1         0        1  
3          1         0        1  
4          1         1        0

c. Use **statsmodels** to instantiate your regression model on the two columns you created in part b., then fit the model using the two columns you created in part **b.** to predict whether or not an individual converts. 

In [37]:
Log = sm.Logit(df2['converted'],df2[['intercept','ab_page']])
results = Log.fit()

Optimization terminated successfully.
         Current function value: 0.366118
         Iterations 6


d. Provide the summary of your model below, and use it as necessary to answer the following questions.

In [38]:
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:              converted   No. Observations:               290584
Model:                          Logit   Df Residuals:                   290582
Method:                           MLE   Df Model:                            1
Date:                Sun, 17 May 2020   Pseudo R-squ.:               8.077e-06
Time:                        23:58:34   Log-Likelihood:            -1.0639e+05
converged:                       True   LL-Null:                   -1.0639e+05
                                        LLR p-value:                    0.1899
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
intercept     -1.9888      0.008   -246.669      0.000      -2.005      -1.973
ab_page       -0.0150      0.011     -1.311      0.190      -0.037       0.007
==============================================================================
"""

e. What is the p-value associated with **ab_page**? Why does it differ from the value you found in **Part II**?<br><br>  **Hint**: What are the null and alternative hypotheses associated with your regression model, and how do they compare to the null and alternative hypotheses in **Part II**?

p-value for ab_page is 0.190, which differs because for A/B testing we are testing the "Null hypothesis" where old page is better or equal to the new page. However on Logistic Regression, we are comparing two side of the coin with equal probability

f. Now, you are considering other things that might influence whether or not an individual converts.  Discuss why it is a good idea to consider other factors to add into your regression model.  Are there any disadvantages to adding additional terms into your regression model?

Advantages:
- one advantage of adding other factors to the system is, looking into multiple factors that might contribute to the result of 'convertion' rate independently.
Disadvantes:
- one dis-advantage of adding other factors to the system is this might cause overlap in collinearity, where we will not be able to see linear reltionship of the variables anymore.
- other factors to consider when adding additional factors to the system is higher order terms, since there could be correlation with one or more x terms we are trying to compare, this also needs to be taken into consideration

g. Now along with testing if the conversion rate changes for different pages, also add an effect based on which country a user lives in. You will need to read in the **countries.csv** dataset and merge together your datasets on the appropriate rows.  [Here](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.join.html) are the docs for joining tables. 

Does it appear that country had an impact on conversion?  Don't forget to create dummy variables for these country columns - **Hint: You will need two columns for the three dummy variables.** Provide the statistical output as well as a written response to answer this question.

In [39]:
df3 = pd.read_csv('countries.csv')
df3.head()

user_id country
0   834778      UK
1   928468      US
2   822059      UK
3   711597      UK
4   710616      UK

In [40]:
df3 = df2.merge(df3, on = "user_id")
df3

user_id                   timestamp      group landing_page  \
0        851104  2017-01-21 22:11:48.556739    control     old_page   
1        804228  2017-01-12 08:01:45.159739    control     old_page   
2        661590  2017-01-11 16:55:06.154213  treatment     new_page   
3        853541  2017-01-08 18:28:03.143765  treatment     new_page   
4        864975  2017-01-21 01:52:26.210827    control     old_page   
5        936923  2017-01-10 15:20:49.083499    control     old_page   
6        679687  2017-01-19 03:26:46.940749  treatment     new_page   
7        719014  2017-01-17 01:48:29.539573    control     old_page   
8        817355  2017-01-04 17:58:08.979471  treatment     new_page   
9        839785  2017-01-15 18:11:06.610965  treatment     new_page   
10       929503  2017-01-18 05:37:11.527370  treatment     new_page   
11       834487  2017-01-21 22:37:47.774891  treatment     new_page   
12       803683  2017-01-09 06:05:16.222706  treatment     new_page   
13       944475  2017-01-22 01:31:09.573836  treatment     new_page   
14       718956  2017-01-22 11:45:11.327945  treatment     new_page   
15       644214  2017-01-22 02:05:21.719434    control     old_page   
16       847721  2017-01-17 14:01:00.090575    control     old_page   
17       888545  2017-01-08 06:37:26.332945  treatment     new_page   
18       650559  2017-01-24 11:55:51.084801    control     old_page   
19       935734  2017-01-17 20:33:37.428378    control     old_page   
20       740805  2017-01-12 18:59:45.453277  treatment     new_page   
21       759875  2017-01-09 16:11:58.806110  treatment     new_page   
22       793849  2017-01-23 22:36:10.742811  treatment     new_page   
23       905617  2017-01-20 14:12:19.345499  treatment     new_page   
24       746742  2017-01-23 11:38:29.592148    control     old_page   
25       892356  2017-01-05 09:35:14.904865  treatment     new_page   
26       773302  2017-01-12 08:29:49.810594  treatment     new_page   
27       913579  2017-01-24 09:11:39.164256    control     old_page   
28       736159  2017-01-06 01:50:21.318242  treatment     new_page   
29       690284  2017-01-13 17:22:57.182769    control     old_page   
...         ...                         ...        ...          ...   
290554   776137  2017-01-12 05:53:12.386730  treatment     new_page   
290555   883344  2017-01-22 23:15:58.645325  treatment     new_page   
290556   825594  2017-01-06 12:37:08.897784  treatment     new_page   
290557   875688  2017-01-14 07:19:49.042869    control     old_page   
290558   927527  2017-01-12 10:52:11.084740    control     old_page   
290559   789177  2017-01-17 18:17:56.215378    control     old_page   
290560   937338  2017-01-19 03:23:22.236666  treatment     new_page   
290561   733101  2017-01-23 12:52:58.711914  treatment     new_page   
290562   679096  2017-01-02 16:43:49.237940  treatment     new_page   
290563   691699  2017-01-09 23:42:35.963486  treatment     new_page   
290564   807595  2017-01-22 10:43:09.285426  treatment     new_page   
290565   924816  2017-01-20 10:59:03.481635    control     old_page   
290566   846225  2017-01-16 15:24:46.705903  treatment     new_page   
290567   740310  2017-01-10 17:22:19.762612    control     old_page   
290568   677163  2017-01-03 19:41:51.902148  treatment     new_page   
290569   832080  2017-01-19 13:18:27.352570    control     old_page   
290570   834362  2017-01-17 01:51:56.106436    control     old_page   
290571   925675  2017-01-07 20:38:26.346410  treatment     new_page   
290572   923948  2017-01-09 16:33:41.104573    control     old_page   
290573   857744  2017-01-05 08:00:56.024226    control     old_page   
290574   643562  2017-01-02 19:20:05.460595  treatment     new_page   
290575   755438  2017-01-18 17:35:06.149568    control     old_page   
290576   908354  2017-01-11 02:42:21.195145    control     old_page   
290577   718310  2017-01-21 22:44:20.378320    control     old_page   
290578   822004  2017-01-04 03:3

In [41]:
df3[['CA','UK','US']] = pd.get_dummies(df3['country'])
df3 = df3.drop(df3['CA'])
df3.head()

user_id                   timestamp      group landing_page  converted  \
2   661590  2017-01-11 16:55:06.154213  treatment     new_page          0   
3   853541  2017-01-08 18:28:03.143765  treatment     new_page          0   
4   864975  2017-01-21 01:52:26.210827    control     old_page          1   
5   936923  2017-01-10 15:20:49.083499    control     old_page          0   
6   679687  2017-01-19 03:26:46.940749  treatment     new_page          1   

   intercept  old_page  ab_page country  CA  UK  US  
2          1         0        1      US   0   0   1  
3          1         0        1      US   0   0   1  
4          1         1        0      US   0   0   1  
5          1         1        0      US   0   0   1  
6          1         0        1      CA   1   0   0

In [42]:
Log_con = sm.Logit(df3['converted'],df3[['intercept','US','UK']])
results = Log_con.fit()
results.summary()

Optimization terminated successfully.
         Current function value: 0.366118
         Iterations 6


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:              converted   No. Observations:               290582
Model:                          Logit   Df Residuals:                   290579
Method:                           MLE   Df Model:                            2
Date:                Sun, 17 May 2020   Pseudo R-squ.:               1.520e-05
Time:                        23:58:39   Log-Likelihood:            -1.0639e+05
converged:                       True   LL-Null:                   -1.0639e+05
                                        LLR p-value:                    0.1984
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
intercept     -2.0375      0.026    -78.364      0.000      -2.088      -1.987
US             0.0408      0.027      1.518      0.129      -0.012       0.094
UK             0.0507      0.028      1.786      0.074      -0.005       0.106
==============================================================================
"""

Since both US, and UK p value is greater than our Type 1 error rate of 0.05. both value doesn't have significant effect of results.

h. Though you have now looked at the individual factors of country and page on conversion, we would now like to look at an interaction between page and country to see if there significant effects on conversion.  Create the necessary additional columns, and fit the new model.  

Provide the summary results, and your conclusions based on the results.

In [43]:
Log_con = sm.Logit(df3['converted'],df3[['intercept','ab_page','US','UK']])
results = Log_con.fit()
results.summary()

Optimization terminated successfully.
         Current function value: 0.366115
         Iterations 6


<class 'statsmodels.iolib.summary.Summary'>
"""
                           Logit Regression Results                           
==============================================================================
Dep. Variable:              converted   No. Observations:               290582
Model:                          Logit   Df Residuals:                   290578
Method:                           MLE   Df Model:                            3
Date:                Sun, 17 May 2020   Pseudo R-squ.:               2.325e-05
Time:                        23:58:41   Log-Likelihood:            -1.0639e+05
converged:                       True   LL-Null:                   -1.0639e+05
                                        LLR p-value:                    0.1757
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
intercept     -2.0300      0.027    -76.248      0.000      -2.082      -1.978
ab_page       -0.0150      0.011     -1.308      0.191      -0.037       0.007
US             0.0408      0.027      1.516      0.129      -0.012       0.093
UK             0.0506      0.028      1.784      0.075      -0.005       0.106
==============================================================================
"""

The results are as stated above, all p - values are less than the type 1 error of 0.05 so it doesn't serve significant impact for the result.

Additional remarks:
1. Based on the result it seems like the old pages lead to higher chance of converting, however I would like the company to increase sample size and the time that user spends on the page, since in practical terms there might be a 'loyal' exisintg users that just might prefer the exisiting page since that is what they are more used to.
2. Also I would focus on specific functionality of the page to see 'WHY' user converted and 'WHY' they didn't

<a id='conclusions'></a>
## Finishing Up

> Congratulations!  You have reached the end of the A/B Test Results project!  You should be very proud of all you have accomplished!

> **Tip**: Once you are satisfied with your work here, check over your report to make sure that it is satisfies all the areas of the rubric (found on the project submission page at the end of the lesson). You should also probably remove all of the "Tips" like this one so that the presentation is as polished as possible.


## Directions to Submit

> Before you submit your project, you need to create a .html or .pdf version of this notebook in the workspace here. To do that, run the code cell below. If it worked correctly, you should get a return code of 0, and you should see the generated .html file in the workspace directory (click on the orange Jupyter icon in the upper left).

> Alternatively, you can download this report as .html via the **File** > **Download as** submenu, and then manually upload it into the workspace directory by clicking on the orange Jupyter icon in the upper left, then using the Upload button.

> Once you've done this, you can submit your project by clicking on the "Submit Project" button in the lower right here. This will create and submit a zip file with this .ipynb doc and the .html or .pdf version you created. Congratulations!

In [44]:
from subprocess import call
call(['python', '-m', 'nbconvert', 'Analyze_ab_test_results_notebook.ipynb'])

0